### PR TITLE
DEV: Add PR prefix labeler and numpy prefix mapping

### PR DESCRIPTION
**Pull request by gerrymanoim**
Awhile ago I made https://github.com/gerrymanoim/pr-prefix-labeler to automatically label PRs based on their prefix (following the numpy guidelines http://www.numpy.org/devdocs/dev/development_workflow.html#writing-the-commit-message, we first used it on https://github.com/quantopian/libpy/pulls?q=is%3Apr+is%3Aclosed).

I figured I'd contribute it back to numpy, if there's any interest. I tried to go through the PRs and fill out the custom mapping (in `.github/pr-prefix-labeler.yml`), apologies if I've missed any.  

Note: Because this action is set to run on `pull_request_target`, it will not run on any PRs until this is merged (see https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request_target for detail).

### CODE REVIEW

**Comment by charris** [comment]
ISTR that there was also an old PR implementing this... cannot find it at the moment.

**Comment by charris** [comment]
The CircleCI failure looks like an old problem that was fixed, might want to rebase on current master.

File: .github/workflows/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+name: "Pull Request Labeler"
+on:
+  pull_request_target:
```
Comment (BvB93) [inline .github/workflows/labeler.yml:3]: Might it be an idea to also add a trigger for [issues](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#issues)?

Comment (seberg) [reply .github/workflows/labeler.yml:3]: Good idea, @gerrymanoim, does that work right now?  I think I would like to try this out and see how it goes.

Comment (eric-wieser) [reply .github/workflows/labeler.yml:3]: Note that with the issue template applying labels already, this may not be that useful for issues.

Comment (seberg) [reply .github/workflows/labeler.yml:3]: Right, I forgot about that, so BUG and ENH are already there, and most of the others apply much less. Lets just try it out, it seems like it is probably nice. Thanks @gerrymanoim!

Comment (gerrymanoim) [reply .github/workflows/labeler.yml:3]: Thanks! I hope it is helpful. I've never tested issues, but could have a look at that.

**Comment by melissawm** [comment]
Thanks @gerrymanoin!

**Pull request merged**